src/rpc/security.rs

```diff
@@ -65,9 +72,10 @@ impl SecurityManager {
         Factory: FnOnce(Channel) -> Client,
     {
         info!("connect to rpc server at endpoint: {:?}", addr);
-        let addr = addr
-            .trim_start_matches("http://")
-            .trim_start_matches("https://");
+
+        let cow_addr = SCHEME_REG.replace(addr, "");
+        let addr: &str = cow_addr.borrow();
```

Comment:
Could you use `&addr` below and avoid the explicit call to `borrow`?

Reply:
done